udata_front/theme/gouvfr/datagouv-components/src/components/Owner/OwnerType.vue

```diff
@@ -0,0 +1,40 @@
+<template>
+  <p class="text-mention-grey fr-text--sm fr-my-0">
+    <OwnerTypeIcon :type="type" /> {{ name }}
+  </p>
+</template>
+<script setup lang="ts">
+import { ref, watchEffect } from "vue";
+import { useI18n } from "vue-i18n";
+import OwnerTypeIcon from "./OwnerTypeIcon.vue";
+import { ASSOCIATION, COMPANY, LOCAL_AUTHORITY, PUBLIC_SERVICE, type OrganizationTypes } from "../../composables/organizations/useOrganizationType";
+import type { UserType } from "../../types/users";
+
+const props = defineProps<{
+  type: OrganizationTypes | UserType;
+}>();
+
+const { t } = useI18n();
+
+const name = ref("");
+
+watchEffect(() => {
+  switch (props.type) {
+    case PUBLIC_SERVICE:
+      name.value = t("Service public");
+      break;
+    case LOCAL_AUTHORITY:
+      name.value = t("Local authority");
+      break;
+    case ASSOCIATION:
+      name.value = t("Association");
+      break;
+    case COMPANY:
+      name.value = t('Company');
+      break;
+    default:
+      name.value = t("User");
+      break;
+  }
```

Comment:
Maybe put `case 'user'` and add a `default` with a never bloc. In my projects I use this:
```
        default:
            throwOnNever(props.type, `Unknown type ${props.type}`);
```

```
export function throwOnNever(_: never, message: string): never {
    throw new Error(message);
}
```

TypeScript will complain if the first argument is not `never` meaning you forget one variant of the enum.